kart/structure.py

```diff
@@ -308,7 +321,6 @@ def check_values_match_schema(self, repo_diff):
                 if new_schema is None:
                     raise InvalidOperation(
                         f"Can't {feature_delta.type} feature {feature_delta.new_key} in deleted dataset {ds_path}",
-                        exit_code=PATCH_DOES_NOT_APPLY,
```

Comment:
why this change?

Reply:
hmm good question. I was looking at some exit-codes to make sure they made sense - eg PATCH_DOES_NOT_APPLY should be for the patch is valid, but doesn't work on HEAD, but there should be different error codes for "can't parse the patch". But this one seems pretty reasonably how it is. I'll revert